Example Site Bar Plot

rdvelazquez opened the pull request:
This PR:
- Renames files and components for consistency
- Organizes the `src` directory 
- Adds two new subcomponents: `BaseSiteBarPlot` and `BaseSiteBarPlotAxis`
- Uses the two new subcomponents in a simple example: `SiteBarPlotExample`
- Updates the webpack config and library export so that alignment.css is exported as a separate file for the library build (see the discussion in [this PR](https://github.com/veg/hyphy-vision/pull/149) from the hyphy-vision repo).

Relates to [this PR](https://github.com/veg/hyphy-gui/pull/24) in the GUI repo.

![alignment-sitebarplotexample](https://user-images.githubusercontent.com/25244432/42326937-89dfd97e-8038-11e8-99ee-796030fd1b55.gif)

stephenshank commented:
This looks awesome! Let me know if you have any questions regarding the scroll broadcaster.

rdvelazquez commented:
> Let me know if you have any questions regarding the scroll broadcaster.

I looked at `ScrollBroadcaster.js` and added some comments for my own understanding (I left most of them in there because I think its useful documentation).

I also found that I had incorrectly referenced the div id of the `BaseSiteBarPlot`; correcting that partially addressed the scrolling issue, so now all the elements (including the `BaseSiteBarPlot` are broadcasting the wheel_event and all the elements (except for the `BaseSiteBarPlot`) are responding to the wheel_event (see below).
![scrolling](https://user-images.githubusercontent.com/25244432/42380021-83983b2a-80fa-11e8-9245-77478b80cf93.gif)
 

I just need to get the `BaseSiteBarPlot` to actually scroll (it _is_ receiving the  wheel_event as evidenced by the console.log statement on line 25 of `BaseSiteBarPlot.jsx`.

I should be able to figure out what's going on and fix it but I'll ping you on slack if I need help. 

p.s. Good work with the ScrollBroadcaster! That was nicely implemented.

rdvelazquez commented:
@stephenshank The scrolling is working. This is now ready for your review.

stephenshank commented on src/components/BaseSiteBarPlot.jsx:
```diff
@@ -0,0 +1,123 @@
+      .enter()
+      .append("g")
+      .attr("transform", function(d, i) {
+        return "translate(" + i * barWidth + ", 10)";
```
`translate(" + (i * barWidth + 1) + ", 10)` would center the bars. Adding `1` since it is half of the offset to the width used below on line 68 below. I've found that it helps my future self to at the very least declare these as named variables, instead of having a bunch of random numbers floating around. Also, why is this 10 here? I see that is necessary but cannot understand why.

rdvelazquez replied:
Good comment! I'm using named variables to make things clearer. 10 is for the top padding and has been named as such.

stephenshank commented on src/components/BaseSiteBarPlot.jsx:
```diff
@@ -0,0 +1,123 @@
+      .attr("height", function(d) {
+        return chart_height - barChartScale(d);
+      })
+      .attr("width", barWidth - 2)
```
May want to replace 2 with a named variable.

stephenshank commented on src/components/BaseSiteBarPlotAxis.jsx:
```diff
@@ -0,0 +1,81 @@
+    axis
+      .append("g")
+      .attr("class", "axis")
+      .attr("transform", `translate(${label_width - 1}, 10)`)
```
Another 10 floating around... is this a padding top?

stephenshank commented on src/components/BaseSiteBarPlotAxis.jsx:
```diff
@@ -0,0 +1,81 @@
+      .append("text")
+      .attr("x", 11)
+      .attr("y", label_width - 20)
+      .attr("transform", `rotate(270 ${0.75 * label_width}, 70)`)
```
Why 70?

rdvelazquez replied:
This may have been for aligning the axis label but it's been replaced by using `text-anchor`.

rdvelazquez commented on src/components/BaseSiteBarPlot.jsx:
```diff
@@ -0,0 +1,123 @@
+
+    var chart = d3
+      .select("#alignmentjs-siteBarPlot")
+      .attr("width", this.props.width)
```
I agree that `width` is used inconstantly and that this inconsistent use could cause problems. Lets look at clear naming convention when we refactor.

rdvelazquez commented on src/components/BaseSiteBarPlotAxis.jsx:
```diff
@@ -0,0 +1,81 @@
+import React from "react";
+const d3 = require("d3");
+
+class BaseSiteBarPlotAxis extends React.Component {
```
It will likely require some tweaking but I think it could be reused. I'm renamed it to `SitePlotAxis`.

rdvelazquez commented:
@stephenshank my last commit addressed all your comments with the exception of:

 - Using css grid
 - Fixing or disabling imports on home page examples
 - Consistently naming/using `width`

I think these are three good items to include in the refactoring that we can do when we make the next site plot example (likely the stacked bar plot for mmvc)
